src/JokeraceEligibility.sol

```diff
@@ -134,53 +144,60 @@ contract JokeraceEligibility is HatsEligibilityModule {
 
   /**
    * @notice Pulls the contest results from the jokerace contest contract.
-   * @dev The eligible wearers for a given completed contest are the top K winners of the contract. In case there is a
-   * tie, meaning that candidates in places K and K+1 have the same score, then the results of this contest rejected.
-   * Additionally, negative scores are also counted as valid scores.
+   * @dev The eligible wearers for a given completed contest are the top K winners of the contest. In case there is a
+   * tie, meaning that candidates in places K and K+1 have the same score, then the results of this contest are
+   * rejected.
    */
   function pullElectionResults() public {
-    GovernorSorting currentContest = GovernorSorting(payable(underlyingContest));
+    GovernorCountingSimple currentContest = GovernorCountingSimple(payable(underlyingContest));
 
-    if (currentContest.state() != IGovernor.ContestState.Completed) {
+    if (currentContest.state() != Governor.ContestState.Completed) {
       revert JokeraceEligibility_ContestNotCompleted();
     }
 
-    // sorted in ascending order
-    uint256[] memory sortedProposalIds = currentContest.sortedProposals(true);
-    uint256 numProposals = sortedProposalIds.length;
-    uint256 numEligibleWearers;
-
-    uint256 k = topK; // save SLOADs
-
-    // check if there's a tie between place k and k + 1. If so, election results are rejected
-    if (numProposals > k) {
-      numEligibleWearers = k;
-      // get the score of candidate in place K
-      uint256 placeK = numProposals - k; // only do this operation once
-      int256 totalVotesPlaceK = getTotalVotes(currentContest, sortedProposalIds[placeK]);
-      // get the score of candidate in place K + 1
-      int256 totalVotesPlaceKPlusOne = getTotalVotes(currentContest, sortedProposalIds[placeK - 1]);
-
-      if (totalVotesPlaceK == totalVotesPlaceKPlusOne) {
-        revert JokeraceEligibility_NoTies();
-      }
-    } else {
-      numEligibleWearers = numProposals;
-    }
-
-    for (uint256 i; i < numEligibleWearers;) {
-      address candidate = getCandidate(currentContest, sortedProposalIds[numProposals - i - 1]);
-      eligibleWearersPerContest[candidate][address(currentContest)] = true;
-
-      // should not overflow based on < numEligibleWearers stopping condition
-      unchecked {
-        ++i;
+    uint256 k = topK;
+    uint256 winningProposalsCount = 0;
+    uint256 currentRank = 1; // ranks start from '1' (the top-rank)
+    bool processed = false;
+    while (!processed) {
```

Comment:
an alternative flow control approach to consider would a for loop with the following structure:

`for (uint256 currentRank = 1; currentRank <= k; ++currentRank) { }`

And then returning true in place of setting `processed = true`;